numba_dpex/core/types/usm_ndarray_type.py

```diff
@@ -66,9 +67,22 @@ def __init__(
         self.device = self.queue.sycl_device.filter_string
 
         if not dtype:
-            dummy_tensor = dpctl.tensor.empty(
-                1, order=layout, usm_type=usm_type, sycl_queue=self.queue
-            )
+            if is_fill_value_float:
```

Comment:
The logic does not belong to the `UsmNdArray` type. It should be moved into the dpnp overloads